apps/frontend/src/components/TopBar.tsx

```diff
@@ -110,15 +104,22 @@ export const TopBar = () => {
   );
 };
 
-const AvgWattText = (props: { currentLap: Lap | null }) => {
-  const { currentLap } = props;
-  if (!currentLap?.normalizedDuration) {
-    return null;
-  }
+const AvgWattText = () => {
+  const { lapData } = useWorkoutState();
+
+  const nonZeroWattLapDatapoints = lapData.filter(
+    (datapoint) => !!datapoint.power
+  );
+
+  const sumWatt = nonZeroWattLapDatapoints.reduce(
+    (totalWatt, datapoint) => totalWatt + (datapoint.power ?? 0),
+    0
+  );
+
   return (
     <Text fontSize={secondaryFontSize}>
       Lap avg:
-      {(currentLap.sumWatt / currentLap.normalizedDuration).toFixed(0)}W
+      {(sumWatt / Math.max(nonZeroWattLapDatapoints.length, 1)).toFixed(0)}W
```

Comment:
Thoughts on doing it this way?

Reply:
I guess it will be fine. Probably worse/possibilty worse for performance. but even a 60 minute lap will only have 3600 datapoints, so it should be fine